From cec0ad6b7a3e8b715db718bfdce3a32289ba2adf Mon Sep 17 00:00:00 2001
From: Marek Vasut <marex@denx.de>
Date: Thu, 30 Nov 2023 22:18:49 +0100
Subject: [PATCH] feat(imx8m): add 3600 MTps DDR PLL rate

Add 3600 MTps DRAM and its 900 MHz PLL setting M=300 P=8 S=0 , so
24 MHz * 300 / 8 / 2^0 = 900 MHz ~ 3600 MTps (x4) .

Upstream-Status: Backport [f1bb459c3192eb6b3fc6b9b77658d82227eae2d5]
Signed-off-by: Marek Vasut <marex@denx.de>
Change-Id: If2743827294efc0f981718f04b772cc462846195
---
 plat/imx/imx8m/ddr/clock.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/plat/imx/imx8m/ddr/clock.c b/plat/imx/imx8m/ddr/clock.c
index 31f2f560e..31b719d28 100644
--- a/plat/imx/imx8m/ddr/clock.c
+++ b/plat/imx/imx8m/ddr/clock.c
@@ -95,6 +95,9 @@ void dram_pll_init(unsigned int drate)
 	case 3732:
 		mmio_write_32(DRAM_PLL_CTRL + 0x4, (311 << 12) | (4 << 4) | 1);
 		break;
+	case 3600:
+		mmio_write_32(DRAM_PLL_CTRL + 0x4, (300 << 12) | (8 << 4) | 0);
+		break;
 	case 3200:
 		mmio_write_32(DRAM_PLL_CTRL + 0x4, (200 << 12) | (3 << 4) | 1);
 		break;
-- 
2.42.0

